Mesh_3/include/CGAL/Mesh_3/polylines_to_protect.h

```diff
@@ -320,15 +333,21 @@ struct Angle_tester
       const typename Kernel::Point_3& p1 = g[v1].point;
       const typename Kernel::Point_3& p2 = g[v2].point;
 
-      if(CGAL::angle(p1, p, p2) == CGAL::ACUTE) {
-        // const typename Kernel::Vector_3 e1 = p1 - p;
-        // const typename Kernel::Vector_3 e2 = p2 - p;
-        // std::cerr << "At point " << p << ": the angle is "
-        //           << ( std::acos(e1 * e2
-        //                          / CGAL::sqrt(e1*e1)
-        //                          / CGAL::sqrt(e2*e2))
-        //                * 180 / CGAL_PI ) << std::endl;
-        return true;
+      if (!m_test_angle)
+      {
+        if (CGAL::angle(p1, p, p2) == CGAL::ACUTE)
+          return true;
+      }
+      else
+      {
+        const typename Kernel::Vector_3 e1 = p1 - p;
+        const typename Kernel::Vector_3 e2 = p2 - p;
+        const double a_rad = std::acos(e1 * e2
+                                      / CGAL::sqrt(e1*e1)
+                                      / CGAL::sqrt(e2*e2));
+        // std::cerr << "At point " << p << ": the angle is " << a << std::endl;
+        if(a_rad < m_angle_rad)
+          return true;
```

Comment:
Here I suggest to use a comparison with the cosine of the angle. Something like:
```c++
  const auto scalar_product = e1 * e2;
  if(CGAL::is_positive(scalar_product)) return true;
  const auto sq_scalar_product = CGAL::square(scalar_product);
  if(sq_scalar_product <= m_angle_sq_cosine * (e1*e1) * (e2*e2)) return true;
```